Source/DafnyCore/Compilers/CSharp/Compiler-Csharp.cs

```diff
@@ -119,8 +130,25 @@ public class CsharpCompiler : SinglePassCompiler {
     }
 
     protected override void EmitBuiltInDecls(SystemModuleManager systemModuleManager, ConcreteSyntaxTree wr) {
-      if (systemModuleManager.MaxNonGhostTupleSizeUsed > 20) {
-        UnsupportedFeatureError(systemModuleManager.MaxNonGhostTupleSizeToken, Feature.TuplesWiderThan20);
+      switch (Options.SystemModuleTranslationMode) {
+        case CommonOptionBag.SystemModuleMode.Omit: {
+            CheckCommonSytemModuleLimits(systemModuleManager);
+            break;
+          }
+        case CommonOptionBag.SystemModuleMode.Populate: {
+            CheckSystemModulePopulatedToCommonLimits(systemModuleManager);
+            break;
+          }
+      }
+
+      // The declarations below would normally be omitted if we aren't compiling the system module,
```

Comment:
Seems like these declarations could be not-internal and only as part of the system module

Reply:
I tried that, but realized `FuncExtensions` in particular is tricky because they are extension methods on the `System.Func` family of delegates, and extension methods only ever apply within the current assembly. Given that I didn't bother spending the time moving the Array helper functions either. I'm adding a comment about this here though.

Reply:
> extension methods only ever apply within the current assembly

I don't get it what you mean. You can use extensions methods defined in a library for sure.